lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -8,6 +8,12 @@ module Execution
       class Base
         STDERR_MESSAGE = 'Command %s completed with the following stderr message: %s'
         VALID_OPTIONS = %i[on_fail expand logger timeout].freeze
+        # Set LC_ALL and LANG to force i18n to C for the duration of this exec;
+        # this ensures that any code that parses the
+        # output of the command can expect it to be in a consistent / predictable format / locale
+        # Set INSIDE_FACTER here to ensure no recursive facter calls are initiated
+        CLI_ENV_VARS = { 'LC_ALL' => 'C', 'LANG' => 'C', 'INSIDE_FACTER' => 'true' }.freeze
```

Comment:
I was kind of assuming the entry point to resolving all facts was in the `FactLoader#load` method so that we wouldn't have to special case this. Or does facter sometimes call execute outside of the `load` method?

Reply:
Ah so environment variables set in the current facter process, e.g. `ENV['INSIDE_FACTER'] = "true", will be inherited by any/all child processes it invokes. That's true for `Open3`, `Kernel.exec`, `system`. One exception is with [`Open3.spawn`](https://docs.ruby-lang.org/en/3.0/Process.html#method-c-spawn), as it's possible to override an environment variable:

```ruby
spawn({"FOO"=>"BAR", "BAZ"=>nil}, command)
```
Or clear environment variables that aren't explicitly specified:
```ruby
spawn({"FOO"=>"BAR"}, command, :unsetenv_others=>true)
```

I'm thinking it'd be clearer to check for `INSIDE_FACTER`, and if it's unset, set it:

```ruby
if ENV['INSIDE_FACTER']
  # we're recursing, skip external facts
else
  ENV['INSIDE_FACTER'] = 'true'
  # call external facts
end
```

Reply:
Wait, so is this conversation more appropriate for the comment [here](https://github.com/puppetlabs/facter/pull/2566#discussion_r1185628488) then?

Reply:
yeah, I'm thinking we shouldn't need to modify any of the execution code, just the part where we load external facts below